test/native/integration-test-helpers/add-block.js

```diff
@@ -44,7 +44,19 @@ export const addBlock = async (
 		},
 	} );
 
-	fireEvent.press( await waitFor( () => getByText( blockName ) ) );
+	const blockButton = await waitFor( () => getByText( blockName ) );
+	// Blocks can perform belated state updates after they are inserted.
+	// To avoid potential `act` warnings, we ensure that all timers and queued
+	// microtasks are executed.
+	await withFakeTimers( async () => {
+		fireEvent.press( blockButton );
+		// Run all timers, in case any performs a state updates.
+		// Column block example: https://t.ly/NjTs
+		act( () => jest.runOnlyPendingTimers() );
+		// Let potential queued microtasks (like Promises) to be executed.
+		// Inner blocks example: https://t.ly/b95nA
+		await act( async () => {} );
```

Comment:
From reviewing the example inner block state changes, it makes sense to abstract this microtask flush, from my perspective. 

The changing state is already abstract, not incredibly block-specific, and does not always result in changes that are queryable in the UI. 

| Setting | Type | Queryable Result |
| - | - | - |
| `allowedBlocks` | `WPBlockType[]` | Seemingly unavailable. |
| `orientation` | `"vertical" \| "horizontal"` | Block mover arrows. |
| `prioritizedInserterBlocks` | `WPBlockType[]` | Seemingly unavailable |
| `templateLock` | `boolean` | Lock icon? |